docs/advanced/fqdn-templating.md

```diff
@@ -0,0 +1,280 @@
+  --fqdn-template="{{.Labels.environment}}.example.com"
+
+# This will result in DNS entries like
+# route53> staging.my-service.example.com
+```
+
+### Multiple FQDN Templates
+
+ExternalDNS allows specifying multiple FQDN templates, which can be useful when you want to create multiple DNS entries for a single service or ingress.
+
+> Be cautious, as this will create multiple DNS records per resource, potentially increasing the number of API calls to your DNS provider.
+
+```yml
+args:
+  --fqdn-template={{.Name}}.example.com,{{.Name}}.svc.example.com
+```
+
+### Conditional Templating combined with Annotations processing
+
+In scenarios where you want to conditionally generate FQDNs based on annotations, you can use Go template functions like or to provide defaults.
+
+```yml
+args:
+  - --combine-fqdn-annotation # this is required to combine FQDN templating and annotation processing
+  - --fqdn-template={{ or .Annotations.dns "invalid" }}.example.com
+  - --exclude-domains=invalid.example.com
+```
+
+### Using Annotations for FQDN Templating]
+
+This example demonstrates how to use annotations in Kubernetes objects to dynamically generate Fully Qualified Domain Names (FQDNs) using the --fqdn-template flag in ExternalDNS.
+
+The Service object includes an annotation dns.company.com/label with the value my-org-tld-v2. This annotation is used as part of the FQDN template to construct the DNS name.
+
+```yml
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-v2
+  namespace: my-namespace
+  annotations:
+    dns.company.com/label: my-org-tld-v2
+spec:
+  type: ClusterIP
+  clusterIP: None
+```
+
+The --fqdn-template flag is configured to use the annotation value (dns.company.com/label) and append the namespace and a custom domain (company.local) to generate the FQDN.
+
+```yml
+args:
+  --source=service
+  --fqdn-template='{{index .ObjectMeta.Annotations "dns.company.com/label"}}.{{ .Namespace }}.company.local'
+
+# For the given Service object, the resulting FQDN will be:
+# route53> my-org-tld-v2.my-namespace.company.local
+```
+
+### DNS Scheme Migration
+
+If you're transitioning from one naming convention to another (e.g., from svc.cluster.local to svc.example.com), --fqdn-template allows you to generate the new records alongside or in place of the old ones — without requiring changes to your Kubernetes manifests.
+
+```yml
+args:
+- --fqdn-template='{{.Name}}.new-dns.example.com'
+```
+
+This helps automate DNS record migration while maintaining service continuity.
+
+### Multi-Variant Domain Support
+
+You can also support regional variants or multi-tenant architectures, where the same service is deployed to different regions or environments:
+
+```yaml
+--fqdn-template='{{ .Name }}.{{ .Environment }}.{{.Region}}.example.com, {{ if eq .Environment "prod" }}{{ .Name }}.my-company.tld{{ end }}'
+```
+
+With additional context (e.g., annotations), this can produce FQDNs like:
+
+```yml
+api.dev.us-east-1.example.com
+api.my-company.tld
```

Comment:
This should be api.prod.us-east-1.example.com?

Reply:
fixed